ui/app/components/pricing-metrics-dates.js

```diff
@@ -0,0 +1,131 @@
+/**
+ * @module PricingMetricsDates
+ * PricingMetricsDates components are used on the Pricing Metrics page to handle queries related to pricing metrics.
+ * This component assumes that query parameters (as in, from route params) are being passed in with the format MM-YYYY,
+ * while the inputs expect a format of MM/YYYY.
+ *
+ * @example
+ * ```js
+ * <PricingMetricsDates @requiredParam={requiredParam} @optionalParam={optionalParam} @param1={{param1}}/>
+ * ```
+ * @param {object} resultStart - resultStart is the start date of the metrics returned. Should be a valid date string that the built-in Date() fn can parse
+ * @param {object} resultEnd - resultEnd is the end date of the metrics returned. Should be a valid date string that the built-in Date() fn can parse
+ * @param {string} [queryStart] - queryStart is the route param (formatted MM-YYYY) that the result will be measured against for showing discrepancy warning
+ * @param {string} [queryEnd] - queryEnd is the route param (formatted MM-YYYY) that the result will be measured against for showing discrepancy warning
+ * @param {number} [defaultSpan=12] - setting for default time between start and end input dates
+ * @param {number} [retentionMonths=24] - setting for the retention months, which informs valid dates to query by
+ */
+import { computed } from '@ember/object';
+import Component from '@ember/component';
+import {
+  compareAsc,
+  differenceInSeconds,
+  isValid,
+  subMonths,
+  startOfToday,
+  format,
+  endOfMonth,
+} from 'date-fns';
+import layout from '../templates/components/pricing-metrics-dates';
+import { parseDateString } from 'vault/helpers/parse-date-string';
+
+export default Component.extend({
+  layout,
+  queryStart: null,
+  queryEnd: null,
+  resultStart: null,
+  resultEnd: null,
+
+  start: null,
+  end: null,
+
+  defaultSpan: 12,
+  retentionMonths: 24,
+
+  startDate: computed('start', function() {
+    if (!this.start) return null;
```

Comment:
nice - I didn't know you didn't have to use function curly braces in this situation?